### ISSUE TITLE
app-agent-config.xml - use of regex to mask segment data

### ISSUE THREAD

**Issue by aleques** [cloudfoundry/java-buildpack#966]
Hi 

there, I'm trying to use match-pattern - regex but it does not work properly.
E.g.:
```xml
        <sensitive-url-filter delimiter="/"
                              segment="3,4,5,6"
                              match-filter="REGEX"
                              match-pattern=":"
                              param-pattern="myParam|myAnotherParam"/>
```

this should mask selected segments that contains `:` but it masks everything.
If I do `match-pattern="="` it works as expected.
This other one does not work as well: `match-pattern="[^a-z¦-]+"`

Is there any subset of regular expression that it's allowed to use here ?

**Comment by dmikusa** [comment]
I don't believe the buildpack is consuming that file, it's just being passed through to the AppDynamics agent. You would need to ask the vendor what they support in that configuration file.

**Comment by aleques** [comment]
> I don't believe the buildpack is consuming that file, it's just being passed through to the AppDynamics agent. You would need to ask the vendor what they support in that configuration file.

Thank you
I'll contact them
